src/utils/helpers/servicesHelpers.ts

```diff
@@ -0,0 +1,63 @@
+import { v4 as uuidv4 } from 'uuid';
+import type { IChart, IData, IDataset } from '../types/api';
+import { ChartParameter } from '../types/enums';
+import type { ObjectType } from '../types/common';
+
+export const formQueryString = (options: IChart, width: number | string, height: number | string): string => {
+  const type = `type:'${options.type}'`;
+
+  const labels = JSON.stringify(options.data.labels.map((label: IData) => label.value));
+  const datasets = JSON.stringify(
+    options.data.datasets.map((dataset: IDataset) => ({
+      label: dataset.label,
+      data: dataset.data.map((dataItem: IData) => dataItem.value),
+    }))
+  )
+    .replace('"label"', 'label')
+    .replace('"data"', 'data');
+  const data = `data:{labels:${labels}, datasets:${datasets}}`;
+
+  const title = options.options.title.text;
+  const optionsURL = `options:{title:{display:true, text:'${title}'}}`;
+
+  const background = `${ChartParameter.background}=transparent`;
+
+  const widthURL = `${ChartParameter.width}=${width}`;
+  const heightURL = `${ChartParameter.height}=${height}`;
+
+  const ratio = `${ChartParameter.ratio}=1`;
+
+  return `?${ChartParameter.chart}={${type},${data},${optionsURL}}&${background}&${widthURL}&${heightURL}&${ratio}`;
```

Comment:
For building such urel I would recommend to use `new URLSearchParams`

Reply:
I have done, but does it look better?

Reply:
In this case you dont need to worry about "&" delimiter